src/goto-harness/function_call_harness_generator.h

```diff
@@ -25,17 +25,17 @@ class function_call_harness_generatort : public goto_harness_generatort
   explicit function_call_harness_generatort(
     ui_message_handlert &message_handler);
   ~function_call_harness_generatort() override;
-  void generate(goto_modelt &goto_model, const irep_idt &harness_function_name)
-    override;
+  void generate(goto_modelt &goto_model) override;
 
 protected:
   void handle_option(
     const std::string &option,
     const std::list<std::string> &values) override;
 
-  void validate_options() override;
+  void validate_options(const goto_modelt &goto_model) override;
```

Comment:
Why are all of these suddenly passed a `goto_model`? And especially `validate_options` shouldn't need one?

Reply:
I guess the question is: Is `validate_options` the right place for the checks?

Reply:
@xbauch It's a reasonable thing to do, we've had this discussion with @danpoe already.